pyalgostrategypool/rsi_macd_crossover_v2.py

```diff
@@ -0,0 +1,80 @@
+        self.main_order_map = None
+
+    def initialize(self):
+        self.main_order_map = {}
+
+    @staticmethod
+    def get_historical_data_duration():
+        return 40
+
+    def get_decision(self, instrument):
+        hist_data = self.get_historical_data(instrument)
+        macdline, macdsignal, _ = talib.MACD(hist_data['close'], fastperiod=self.timeperiod_fast, slowperiod=self.timeperiod_slow, signalperiod=self.timeperiod_signal)
+        rsi_value = talib.RSI(macdsignal, timeperiod=self.timeperiod_rsi)
+
+        oversold_list = [self.oversold_value] * rsi_value.size
+        overbought_list = [self.overbought_value] * rsi_value.size
+
+        oversold_crossover_value = self.utils.crossover(rsi_value, oversold_list)
+        overbought_crossover_value = self.utils.crossover(rsi_value, overbought_list)
+
+        return oversold_crossover_value, overbought_crossover_value
+
+    def strategy_select_instruments_for_entry(self, candle, instruments_bucket):
+        instruments, meta = [], []
+
+        for instrument in instruments_bucket:
+            if self.main_order_map.get(instrument) is None:
+                oversold_crossover_value, overbought_crossover_value = self.get_decision(instrument)
+                if oversold_crossover_value == 1:
+                    instruments.append(instrument)
+                    meta.append({'action': 'BUY'})
+                elif overbought_crossover_value == -1:
+                    instruments.append(instrument)
+                    meta.append({'action': 'SELL'})
+
+        return instruments, meta
+
+    def strategy_enter_position(self, candle, instrument, meta):
+        self.main_order_map[instrument] = _ = self.broker.OrderRegular(instrument, meta['action'], quantity=self.number_of_lots * instrument.lot_size)
+        return _
+
+    def strategy_select_instruments_for_exit(self, candle, instruments_bucket):
+        instruments, meta = [], []
+
+        for instrument in instruments_bucket:
+            if self.main_order_map.get(instrument) is not None:
+                oversold_crossover_value, overbought_crossover_value = self.get_decision(instrument)
+                if (oversold_crossover_value == -1) and self.main_order_map[instrument].order_transaction_type.value == 'SELL':
+                    instruments.append(instrument)
+                    meta.append({'action': 'EXIT'})
+                elif (overbought_crossover_value == 1) and self.main_order_map[instrument].order_transaction_type.value == 'BUY':
+                    instruments.append(instrument)
+                    meta.append({'action': 'EXIT'})
+
+        return instruments, meta
+
+    def strategy_exit_position(self, candle, instrument, meta):
+        if meta['action'] == 'EXIT':
+            self.main_order_map[instrument].exit_position()
+            self.main_order_map[instrument] = None
+            return True
+
+        return False
```

Comment:
O-I-L not done, do it for all strategies

Reply:
Done